Basis/Packages/com.basis.framework/Camera/Helpers/BasisDepthOfFieldInteractionHandler.cs

```
@@ -0,0 +1,64 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+[System.Serializable]
+public class BasisDepthOfFieldInteractionHandler : MonoBehaviour
+{
+    [Header("References")]
+    public BasisHandHeldCamera cameraController;
+    public RectTransform focusCursor;
+    public Toggle depthOfFieldToggle;
+
+    [Header("Raycasting")]
+    public float maxRaycastDistance = 1000f;
+    private void OnEnable()
+    {
+        if (depthOfFieldToggle != null)
+            depthOfFieldToggle.onValueChanged.AddListener(SetCursorVisibility);
+    }
+    public void SetDoFState(bool enabled)
+    {
+        if (cameraController?.MetaData?.depthOfField != null)
```

your using ? ???????? a lot instead lets just not.

if we get a null ref we should just error instead of silently continuing.